src/object-store/src/util/thread_id.hpp

```diff
@@ -0,0 +1,104 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#ifndef REALM_THREAD_ID_HPP
+#define REALM_THREAD_ID_HPP
+
+#include <realm/util/features.h>
+
+#include <atomic>
+
+#if __has_feature(tls) || __has_feature(cxx_thread_local)
+#define REALM_HAS_THREAD_LOCAL 1
+#else
+#define REALM_HAS_THREAD_LOCAL 0
+#include <pthread.h>
+#endif
+
+namespace realm {
+
+using thread_id_t = std::size_t;
+
+namespace _impl {
```

Comment:
The other object store code base in headers is there because of the use of
templates. Anything more than a few lines should go in a cpp file when
possible.
On Thu, Jun 9, 2016 at 15:15 Mark Rowe <notifications@github.com> wrote:

> In src/object-store/src/util/thread_id.hpp
> <https://github.com/realm/realm-js/pull/480#discussion_r66531768>:
>
> > +#include <realm/util/features.h>
> > +
> > +#include <atomic>
> > +
> > +#if __has_feature(tls) || __has_feature(cxx_thread_local)
> > +#define REALM_HAS_THREAD_LOCAL 1
> > +#else
> > +#define REALM_HAS_THREAD_LOCAL 0
> > +#include <pthread.h>
> > +#endif
> > +
> > +namespace realm {
> > +
> > +using thread_id_t = std::size_t;
> > +
> > +namespace _impl {
>
> I didn't expect this to be a contentious point. If it were only the ~5
> line function itself I'd agree it's not worth it, but the fact there's 10x
> that in supporting code is why i suggested moving it to a .cpp file.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/realm/realm-js/pull/480/files/563596ca9386503ba09938965362065e1bf2cfad#r66531768>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAPRIOth4u6Ds3IuuGfUTfvoKVbHsgSKks5qKJBugaJpZM4Ixjq7>
> .
>